ani-cli-win

```diff
@@ -72,32 +72,48 @@ get_dpage_link() {
 
 	# credits to fork: https://github.com/Dink4n/ani-cli for the fix
 	curl -4 -s "$base_url/$anime_id-episode-$ep_no" |
-	sed -n -E '
-		/^[[:space:]]*<li class="dowloads">/{
-		s/.*href="([^"]*)".*/\1/p
-		q
-		}'
+    sed -n -E 's/^[[:space:]]*<a href="#" rel="100" data-video="([^"]*)".*/\1/p' |
+	    sed 's/^/https:/g'
+}
+
+decrypt_link() {
+    ajax_url='https://gogoplay.io/encrypt-ajax.php'
+
+    #get the id from the url
+    video_id=$(printf "$1" | cut -d\? -f2 | cut -d\& -f1 | sed 's/id=//g')
+
+    #construct ajax parameters
+    secret_key='3235373436353338353932393338333936373634363632383739383333323838'
+    iv='34323036393133333738303038313335'
+    ajax=$(printf "$video_id" | openssl enc -aes256  -K "$secret_key" -iv "$iv" -a)
+
+    #send the request to the ajax url
+    curl -s -H 'x-requested-with:XMLHttpRequest' "$ajax_url" -d "id=$ajax" -d "time=69420691337800813569" \
+		| tr '"' '\n' \
+		| sed -n -E 's/.*cdn.*/\0/p' \
+		| sed 's/\\//g' \
+		| sed '/.vtt/d'
 }
 
 get_video_quality() {
 	dpage_url=$1
 
-	video_links=$(curl -4 -s "$dpage_url" | sed -n -E 's/.*href="([^"]*)" download>Download.*/\1/p' | sed 's/amp;//')
+	video_links=$(decrypt_link "$dpage_url")
 	case $quality in
 		best)
-			video_link=$(printf '%s' "$video_links" | tail -n 1)
+			video_link=$(printf '%s' "$video_links" | head -n 4 | tail -n 1)
 			;;
 
 		worst)
 			video_link=$(printf '%s' "$video_links" | head -n 1)
 			;;
 
 		*)
-			video_link=$(printf '%s' "$video_links" | grep -i "${quality}p")
+			video_link=$(printf '%s' "$video_links" | grep -i "${quality}p" | head -n 1)
 			if [ -z "$video_link" ]; then
-				printf "$c_red%s$c_reset\n" "Current video quality is not available (defaulting to highest quality)" >&2
+				err "Current video quality is not available (defaulting to best quality)"
 				quality=best
-				video_link=$(printf '%s' "$video_links" | tail -n 1)
+				video_link=$(printf '%s' "$video_links" | head -n 4 | tail -n 1)
```

Comment:
Why does this need changing? Is the video link now always in line 4, with padding below sometimes?

Reply:
Oh, this is just backporting fixes to the windows branch. My bad